lib/matplotlib/axes/_axes.py

```diff
@@ -5549,6 +5549,15 @@ def _pcolorargs(funcname, *args, shading='flat'):
         # - reset shading if shading='auto' to flat or nearest
         #   depending on size;
 
+        _valid_shading = ['gouraud', 'nearest', 'flat', 'auto']
+        try:
+            cbook._check_in_list(_valid_shading, shading=shading)
```

Comment:
Maybe its overkill, but should this be done w/ `normalize_kwargs` to be consistent w/ other kwarg checks?  Otherwise this looks good to me.

Reply:
I don't think `normalize_kwargs` would work, as it's not part of the kwargs given to an `Artist`.